server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndex.java

```diff
@@ -49,42 +53,59 @@
 import static org.sonar.server.component.index.ComponentIndexDefinition.FIELD_NAME;
 import static org.sonar.server.component.index.ComponentIndexDefinition.FIELD_QUALIFIER;
 import static org.sonar.server.component.index.ComponentIndexDefinition.INDEX_TYPE_COMPONENT;
+import static org.sonar.server.component.index.ComponentIndexDefinition.NAME_ANALYZERS;
 
 public class ComponentIndex {
 
   private static final String FILTERS_AGGREGATION_NAME = "filters";
   private static final String DOCS_AGGREGATION_NAME = "docs";
 
   private final EsClient client;
+  private final FavoriteFinder favoriteFinder;
```

Comment:
Done.

I had changed it back and forth, finally decided to keep it in ComponentIndex, so that all potential users of the ComponentIndex can share the implementation.

Now it is back in SuggestionsAction again, making favorites and recentlyBrowsed more similar, which is a good thing, too.